Content.Server/Commands/Chat/SuicideCommand.cs

```diff
@@ -121,9 +124,9 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
 
             dmgComponent.SetDamage(DamageType.Piercing, 200, owner);
 
-            // Prevent the player from returning to the body. Yes, this is an ugly hack.
-            var ghost = new Ghost(){CanReturn = false};
-            ghost.Execute(shell, argStr, Array.Empty<string>());
+            // Prevent the player from returning to the body.
+            // Note that mind cannot be null because otherwise owner would be null.
+            IoCManager.Resolve<IGameTicker>().OnGhostAttempt(mind!, false);
```

Comment:
Discussed this with @Zumorica on Discord and it'd be better to replace calls to OnGhostAttempt with an EntityEvent if possible.

Reply:
I think that's out of scope for this PR unless someone wants to give me an idea of where everything actually goes in this regard

Reply:
That's fine. Maybe just create an issue for that then.

Reply:
https://github.com/space-wizards/space-station-14/issues/4112